core/packages/contracts/test/apps/xcm_app.test.ts

```diff
@@ -0,0 +1,206 @@
+import { defaultAbiCoder } from "@ethersproject/abi"
+import { expect, loadFixture } from "../setup"
+import { xcmAppFixture } from "./fixtures"
+import { XcmFungibleAsset__factory } from "../../src"
+
+let PARA_2001_ORIGIN = "0x2001000000000000000000000000000000000000000000000000000000000000"
+let PARA_2002_ORIGIN = "0x2002000000000000000000000000000000000000000000000000000000000000"
+
+describe("XCMApp", function () {
```

Comment:
We gonna migrate to Foundry, so I'd pause writing hardhat tests until then (and when our design is more concrete)